### ISSUE TITLE
Reboot stuck on Raspbian Stretch

### ISSUE THREAD

**Issue by knumskull** [ansible/ansible#46562]
##### SUMMARY
Tried the new reboot feature in ansible 2.7. But ansible prompt sticks, while system is already rebooted.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

reboot

##### ANSIBLE VERSION

```paste below
ansible 2.7.0
  config file = None
  configured module search path = [u'/Users/steffen/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Library/Python/2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.10 (default, Feb  7 2017, 00:08:15) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)]
```

##### CONFIGURATION
freshly installed ansible 2.7 without any modification

##### OS / ENVIRONMENT

raspbian stretch

```
$ uname -a
Linux pihab.home.wlan 4.14.34-v7+ #1110 SMP Mon Apr 16 15:18:51 BST 2018 armv7l GNU/Linux

$ timedatectl status
      Local time: Sat 2018-10-06 07:09:33 CEST
  Universal time: Sat 2018-10-06 05:09:33 UTC
        RTC time: n/a
       Time zone: Europe/Berlin (CEST, +0200)
 Network time on: yes
NTP synchronized: yes
 RTC in local TZ: no
```

##### STEPS TO REPRODUCE

Simply run adhoc `reboot`command 

```bash
$ ansible -b -i hosts pihab -m reboot
```

##### EXPECTED RESULTS

Prompt should return after system successfully rebooted

##### ACTUAL RESULTS

On executing the reboot command, the prompt will not return without manual interrupt.

It seems the reboot-check failing.

Code will check for `1970-01-01 00:00`, but system return `1970-01-01 01:00`(different time)

```
$ who -b
         system boot  1970-01-01 01:00
```

```paste below
reboot: attempting to get system boot time
<pihab.home.wlan> ESTABLISH SSH CONNECTION FOR USER: pi
<pihab.home.wlan> SSH: EXEC ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=pi -o ConnectTimeout=10 -o ControlPath=/Users/steffen/.ansible/cp/63197cf86e -tt pihab.home.wlan '/bin/sh -c '"'"'sudo -H -S -n -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-amrletdwexioafvfibsiorkmauqmkyzi; who -b'"'"'"'"'"'"'"'"' && sleep 0'"'"''
Escalation succeeded
<pihab.home.wlan> (0, '         system boot  1970-01-01 01:00\r\n', 'OpenSSH_7.4p1, LibreSSL 2.5.0\r\ndebug1: Reading configuration data /Users/steffen/.ssh/config\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 3926\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\nShared connection to pihab.home.wlan closed.\r\n')
```

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/system/reboot.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/system/reboot.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ssiegel** [comment]
I experience this issue on Raspbian Stretch as well, but for me the output of `who -b` is `system boot  Jan  1 00:00`. This date/time format seems to be used in the `C` or `POSIX` locale, which means it also happens on a freshly installed system having no locale set. (see https://github.com/coreutils/coreutils/blob/master/src/who.c#L805)

**Comment by ssiegel** [comment]
Why is /proc/sys/kernel/random/boot_id called a “last resort” in https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/action/reboot.py#L122? According to the random(4) manpage it is available since kernel 2.3.16 (released in 1999), so it should be pretty safe to use as first choice on Linux.

**Comment by samdoran** [comment]
@ssiegel I didn't know about `/proc/sys/kernel/random/boot_id` when I started writing this module. I was thinking of switching this to the primary option, but had not found when exactly it was added to the kernel. Thanks for finding that!

I have a WIP PR [#45656](https://github.com/ansible/ansible/pull/45656/files#diff-d8e166386a38471609399c2c6fe07ba5R39) that does switch to using this method primarily, but that includes a bunch of other changes as well.
